src/main/java/org/killbill/billing/plugin/helloworld/HelloWorldInvoicePlugin.java

```diff
@@ -0,0 +1,67 @@
+package org.killbill.billing.plugin.helloworld;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import org.killbill.billing.account.api.Account;
+import org.killbill.billing.invoice.api.Invoice;
+import org.killbill.billing.invoice.api.InvoiceItem;
+import org.killbill.billing.notification.plugin.api.ExtBusEvent;
+import org.killbill.billing.osgi.libs.killbill.OSGIConfigPropertiesService;
+import org.killbill.billing.osgi.libs.killbill.OSGIKillbillAPI;
+import org.killbill.billing.osgi.libs.killbill.OSGIKillbillEventDispatcher.OSGIKillbillEventHandler;
+import org.killbill.billing.payment.api.PluginProperty;
+import org.killbill.billing.plugin.api.invoice.PluginInvoiceItem;
+import org.killbill.billing.plugin.api.invoice.PluginInvoicePluginApi;
+import org.killbill.billing.util.callcontext.CallContext;
+import org.killbill.billing.util.callcontext.TenantContext;
+import org.killbill.clock.Clock;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+class HelloWorldInvoicePlugn extends PluginInvoicePluginApi  implements OSGIKillbillEventHandler {
+	
+	public HelloWorldInvoicePlugn(OSGIKillbillAPI killbillAPI, OSGIConfigPropertiesService configProperties, Clock clock) {
+		super(killbillAPI, configProperties, clock);
+		// TODO Auto-generated constructor stub
+	}
+
+	@Override
+	public List<InvoiceItem> getAdditionalInvoiceItems(final Invoice newInvoice, final boolean dryRun, 
+			final Iterable<PluginProperty> properties, final CallContext callContext) {
+		UUID accountId = newInvoice.getAccountId();
+        Account account = getAccount(accountId, callContext);
+		Set<Invoice> invoices =  allInvoicesOfAccount(account, newInvoice, callContext);
+		
+		ImmutableList.Builder<InvoiceItem> additionalItems  = ImmutableList.builder();
+		for(Invoice invoice : invoices) {
+			if (invoice.equals(newInvoice)) {
+                ImmutableList.Builder<InvoiceItem> newIntItems = ImmutableList.builder();
```

Comment:
If at all it is necessary to retrieve account invoices and check for a match, I would suggest using the `invoiceId` (`invoice.getId.equals(newInvoice.getId()`)

Reply:
Also, after a match is found, you can `break` out of the `for` loop as there will only be one invoice with the same id.